### PR TITLE
@kanaabe => Update React to 15

### PR DESCRIPTION
**Pull request by eessex**
- Update all render functions to use ReactDOM.render
- Remove all instances of getDOMNode
- Add display names (for easy debugging)
- Add keys when erroring
- Update all tests to use reactTestUtils, reactDOMServer for rendering

There are still two tests in here that aren't passing, will wrap those up soon

### CODE REVIEW

**Comment by kanaabe** [comment]
Woo very exciting! 

It looks like the shrinkwrap was removed. I'm hesitant to re-shrink everything because of those memory leak issues we had a while back so I think the best bet is to copy over the current shrinkwrap and `npm install <pkg> --save` each package individually. Probably could even be --save-dev for the test utils.

File: client/apps/edit/components/section_container/index.coffee [hunk 1]
```diff
@@ -61,6 +63,7 @@ module.exports = React.createClass
           when 'fullscreen' then SectionFullscreen
           when 'callout' then SectionCallout
           when 'toc' then SectionToc
+          when 'image' then SectionImage
```
Comment (kanaabe) [inline client/apps/edit/components/section_container/index.coffee]: Ah see line 18 and 69, I've added these already from [this PR](https://github.com/artsy/positron/pull/917/commits/917cf32ef04dafd4313cd445b5092a4643ed81a1).

File: client/apps/edit/components/section_embed/index.coffee [hunk 2]
```diff
@@ -8,6 +8,7 @@ sd = require('sharify').data
 { div, section, label, nav, input, a, h1, p, strong, span, form, button, iframe } = React.DOM
 
 module.exports = React.createClass
+  displayName: 'SectionEmbed'
```
Comment (kanaabe) [inline client/apps/edit/components/section_embed/index.coffee]: Thanks for adding these!

File: client/apps/edit/components/section_embed/index.coffee [hunk 3]
```diff
@@ -26,9 +27,9 @@ module.exports = React.createClass
 
   submitUrl: (e) ->
     e.preventDefault()
-    url = @refs.url.getDOMNode().value
-    height = @refs.height.getDOMNode().value
-    mobileHeight = @refs.mobileHeight.getDOMNode().value
+    url = @refs.url.value
+    height = @refs.height.value
+    mobileHeight = @refs.mobileHeight.value
```
Comment (kanaabe) [inline client/apps/edit/components/section_embed/index.coffee]: So much cleaner ✨

**Comment by kanaabe** [comment]
Very awesome work here! After resolving that one dupe comment, this is ready to be merged.

**Comment by eessex** [comment]
Cool! Thanks for noticing the duplicates -- didn't see them when I pulled changes in -- gonna go ahead and merge when this turns green :)

**Pull request merged**